CMakeLists.txt

```diff
@@ -30,3 +30,11 @@ endif()
 if(CONFIG_LITTLEFS_MULTIVERSION)
     target_compile_definitions(${COMPONENT_LIB} PUBLIC -DLFS_MULTIVERSION)
 endif()
+
+if(CONFIG_LITTLEFS_MALLOC_STRATEGY_DISABLE)
+    target_compile_definitions(${COMPONENT_LIB} PUBLIC -DLFS_NO_MALLOC)
+endif()
+
+if(NOT CONFIG_LITTLEFS_ASSERTS)
+    target_compile_definitions(${COMPONENT_LIB} PUBLIC -DLFS_NO_ASSERT)
+endif()
```

Comment:
Good spot on the asserts, since `heap_caps_malloc` guarantees 32-bit alignment we can drop the use of `heap_caps_aligned_alloc`.

This also explains why using the system `malloc( )` was working fine before, since under the hood ESP-IDF calls `heap_caps_malloc_default( )`

Reply:
> so I'm honestly pretty bad at CMake. Is it appropriate to have these options both here as well as in KConfig?

CMake I have only absorbed partially by painful osmosis :)

The Kconfig sets the scene for our `CMakeLists.txt` and the esp_littlefs sources, but the upstream littlefs still relies on the original flags being set (which we can automatically derive from Kconfig).

The net result is that the user ticks pretty Kconfig boxes, we use those options directly in our code, and automatically set the right things for littlefs upstream.

Reply:
Looking at where `LFS_NO_ASSERT` is used in `lfs.c`, it's impact is minimal - it prevents compiling of `lfs_mlist_isopen( )` which is only used by asserts and would probably be optimised away by the linker anyway.  So strictly we don't _need_ to set `LFS_NO_ASSERT` using CMake, but it's still technically the correct thing to do it may get used in other ways for future iterations of upstream littlefs.